Rust binding in Cryptography 3.4+ dynamically links with non-existent library on macOS

Issue cloned from the Cryptography issue tracker per request, as setuptools-rust handles this for them: https://github.com/pyca/cryptography/issues/5807

The Cryptography wheel for macOS includes _rust.abi3.so. This references a library that isn't shipped and has a path clearly from the build machine:
```
(foo) dpage@hal:~/git/pgadmin4/foo/lib/python3.8/site-packages/cryptography/hazmat/bindings$ otool -L _rust.abi3.so 
_rust.abi3.so:
	/Users/runner/work/cryptography/cryptography/cryptography-3.4.4/src/rust/target/release/deps/libcryptography_rust.dylib (compatibility version 0.0.0, current version 0.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1281.100.1)
	/usr/lib/libresolv.9.dylib (compatibility version 1.0.0, current version 1.0.0)
```

Whilst this may not break the package, it:

A) Is not good practice
B) Breaks build systems that automatically fixup library paths when creating relocatable appbundles.

B is the issue I'd like to see fixed, as the current situation requires me to modify code that's worked for many years (15+) to ignore errors when processing this particular file.

An additional issue, again likely benign for the general use case, is that _rust.abi3.so does not have the executable bit set as is generally the norm for shared libraries on *nix platforms. This also required me to change build code, as searches for binaries that need to be codesigned filtered possible targets based on the executable bit, before doing further tests to see if the file was actually an executable or library (which gets *very* expensive without that check, in a codebase with thousands of Python/JS/HTML files etc).

Probably not relevant, but here's my system info:
```
(foo) dpage@hal:~/git/pgadmin4/foo/lib/python3.8/site-packages/cryptography/hazmat/bindings$ pip freeze
cffi==1.14.4
cryptography==3.4.4
pycparser==2.20
(foo) dpage@hal:~/git/pgadmin4/foo/lib/python3.8/site-packages/cryptography/hazmat/bindings$ python --version
Python 3.8.2
(foo) dpage@hal:~/git/pgadmin4/foo/lib/python3.8/site-packages/cryptography/hazmat/bindings$ pip --version
pip 21.0.1 from /Users/dpage/git/pgadmin4/foo/lib/python3.8/site-packages/pip (python 3.8)
(foo) dpage@hal:~/git/pgadmin4/foo/lib/python3.8/site-packages/cryptography/hazmat/bindings$ sw_vers -productVersion
11.1
```

Thanks!

I can't get it to build unfortunately. It looks like the linker is failing to find what I assume are functions in the Python library, however even explicitly adding that to LDFLAGS isn't helping (in fact it looks like LDFLAGS is being ignored):
```
   Compiling cryptography-rust v0.1.0 (/Users/dpage/git/cryptography/src/rust)
error: linking with `/usr/bin/clang` failed: exit code: 1
  |
  = note: "/usr/bin/clang" "-m64" "-arch" "x86_64" "-L" "/opt/local/lib/rustlib/x86_64-apple-darwin/lib" "/Users/dpage/git/cryptography/src/rust/target/debug/deps/cryptography_rust.15gjghlu49iac21.rcgu.o" "/Users/dpage/git/cryptography/src/rust/target/debug/deps/cryptography_rust.1jkgirnp9b8rrlvu.rcgu.o" "/Users/dpage/git/cryptography/src/rust/target/debug/deps/cryptography_rust.1tk8h0snkgefotww.rcgu.o" "/Users/dpage/git/cryptography/src/rust/target/debug/deps/cryptography_rust.2aqkg278dbcjwmv6.rcgu.o" "/Users/dpage/git/cryptography/src/rust/target/debug/deps/cryptography_rust.2utiffjkt71shkdn.rcgu.o" "/Users/dpage/git/cryptography/src/rust/target/debug/deps/cryptography_rust.36tytxljafqzsxvl.rcgu.o" "/Users/dpage/git/cryptography/src/rust/target/debug/deps/cryptography_rust.3q3o6svhhkajtmx3.rcgu.o" "/Users/dpage/git/cryptography/src/rust/target/debug/deps/cryptography_rust.3z005hih81m7u173.rcgu.o" "/Users/dpage/git/cryptography/src/rust/target/debug/deps/cryptography_rust.41f5nnxh5akbe6su.rcgu.o" "/Users/dpage/git/cryptography/src/rust/target/debug/deps/cryptography_rust.4g8xce78q4nsgvw1.rcgu.o" "/Users/dpage/git/cryptography/src/rust/target/debug/deps/cryptography_rust.4npvpg2s7ofjp0zy.rcgu.o" "/Users/dpage/git/cryptography/src/rust/target/debug/deps/cryptography_rust.52fyt1allp219wbc.rcgu.o" "/Users/dpage/git/cryptography/src/rust/target/debug/deps/cryptography_rust.53nku9jpzl7eq0ib.rcgu.o" "/Users/dpage/git/cryptography/src/rust/target/debug/deps/cryptography_rust.54gxbp9era3lo0fz.rcgu.o" "/Users/dpage/git/cryptography/src/rust/target/debug/deps/cryptography_rust.55pzwx2vvhn527gw.rcgu.o" "/Users/dpage/git/cryptography/src/rust/target/debug/deps/cryptography_rust.593gs740su5s2o7h.rcgu.o" "/Users/dpage/git/cryptography/src/rust/target/debug/deps/cryptography_rust.5d3q7cg7uuuo8mx.rcgu.o" "/Users/dpage/git/cryptography/src/rust/target/debug/deps/cryptography_rust.5flweajuvviuskww.rcgu.o" "/Users/dpage/git/cryptography/src/rust/target/debug/deps/cryptography_rust.8p5grgilkxpgbjr.rcgu.o" "/Users/dpage/git/cryptography/src/rust/target/debug/deps/cryptography_rust.dqvfq4dax0kjc6d.rcgu.o" "/Users/dpage/git/cryptography/src/rust/target/debug/deps/cryptography_rust.u68ppxlip0v5bb1.rcgu.o" "-o" "/Users/dpage/git/cryptography/src/rust/target/debug/deps/libcryptography_rust.dylib" "-Wl,-exported_symbols_list,/var/folders/48/dq6ggtt92l95nzjj_4xqrd400000gn/T/rustc3A6XfP/list" "/Users/dpage/git/cryptography/src/rust/target/debug/deps/cryptography_rust.205mtonn7kjxl86n.rcgu.o" "-Wl,-dead_strip" "-dynamiclib" "-Wl,-dylib" "-nodefaultlibs" "-L" "/Users/dpage/git/cryptography/src/rust/target/debug/deps" "-L" "/opt/local/lib/rustlib/x86_64-apple-darwin/lib" "/Users/dpage/git/cryptography/src/rust/target/debug/deps/libpyo3-e06e690e5cecc294.rlib" "/Users/dpage/git/cryptography/src/rust/target/debug/deps/libparking_lot-1ca969a9be3617d0.rlib" "/Users/dpage/git/cryptography/src/rust/target/debug/deps/libparking_lot_core-ab5e826405a87dfb.rlib" "/Users/dpage/git/cryptography/src/rust/target/debug/deps/libsmallvec-50db1cb0396c8bf7.rlib" "/Users/dpage/git/cryptography/src/rust/target/debug/deps/liblock_api-18e840c0f9e7a2f8.rlib" "/Users/dpage/git/cryptography/src/rust/target/debug/deps/libscopeguard-9e058c2a551879f8.rlib" "/Users/dpage/git/cryptography/src/rust/target/debug/deps/libinstant-f369cae457cc2e1b.rlib" "/Users/dpage/git/cryptography/src/rust/target/debug/deps/libcfg_if-2c21884bff284602.rlib" "/Users/dpage/git/cryptography/src/rust/target/debug/deps/liblibc-95a48ad04b2649c9.rlib" "/Users/dpage/git/cryptography/src/rust/target/debug/deps/libunindent-faf2e9e2e0d84eed.rlib" "/Users/dpage/git/cryptography/src/rust/target/debug/deps/libinventory-0126821be40f290e.rlib" "/opt/local/lib/rustlib/x86_64-apple-darwin/lib/libstd-00c1b2a30df18801.rlib" "/opt/local/lib/rustlib/x86_64-apple-darwin/lib/libpanic_unwind-f512d396ad36f67e.rlib" "/opt/local/lib/rustlib/x86_64-apple-darwin/lib/libobject-fbd5b42f6031d7ca.rlib" "/opt/local/lib/rustlib/x86_64-apple-darwin/lib/libaddr2line-bed1ac853b0deef6.rlib" "/opt/local/lib/rustlib/x86_64-apple-darwin/lib/libgimli-01136bd058fa514d.rlib" "/opt/local/lib/rustlib/x86_64-apple-darwin/lib/librustc_demangle-200142ba51d1bfd5.rlib" "/opt/local/lib/rustlib/x86_64-apple-darwin/lib/libhashbrown-097526c57563a37d.rlib" "/opt/local/lib/rustlib/x86_64-apple-darwin/lib/librustc_std_workspace_alloc-858253b43fca2811.rlib" "/opt/local/lib/rustlib/x86_64-apple-darwin/lib/libunwind-78e01da90642f7b0.rlib" "/opt/local/lib/rustlib/x86_64-apple-darwin/lib/libcfg_if-1ccf017aaea7a930.rlib" "/opt/local/lib/rustlib/x86_64-apple-darwin/lib/liblibc-20d74e683e6fc577.rlib" "/opt/local/lib/rustlib/x86_64-apple-darwin/lib/liballoc-4d8fa99874399801.rlib" "/opt/local/lib/rustlib/x86_64-apple-darwin/lib/librustc_std_workspace_core-0387df0e85a6becc.rlib" "/opt/local/lib/rustlib/x86_64-apple-darwin/lib/libcore-86974b7b6295ebe8.rlib" "/opt/local/lib/rustlib/x86_64-apple-darwin/lib/libcompiler_builtins-3847af7e923e2fe0.rlib" "-lSystem" "-lresolv" "-lc" "-lm"
  = note: Undefined symbols for architecture x86_64:
            "_PyExc_BaseException", referenced from:
                _$LT$pyo3..exceptions..PyBaseException$u20$as$u20$pyo3..type_object..PyTypeInfo$GT$::type_object_raw::hc392a0b462685cbe in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.0.rcgu.o)
            "_PyUnicode_AsEncodedString", referenced from:
                pyo3::types::string::PyString::to_string_lossy::h5e131baedd19615b in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.14.rcgu.o)
            "_PyUnicode_FromStringAndSize", referenced from:
                pyo3::types::string::PyString::new::hdd1b5d2a5fd89740 in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.14.rcgu.o)
            "_PyObject_Repr", referenced from:
                pyo3::types::any::PyAny::repr::h009e16ca240e83e1 in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.15.rcgu.o)
            "_PyUnicode_AsUTF8AndSize", referenced from:
                pyo3::types::string::PyString::to_str::hc9fdbe9616455bd2 in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.14.rcgu.o)
            "_PyObject_SetAttr", referenced from:
                pyo3::types::any::PyAny::setattr::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h326355d88ab1caa1 in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.15.rcgu.o)
                pyo3::types::any::PyAny::setattr::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h54825ade16374a9d in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.15.rcgu.o)
            "_PyErr_NewException", referenced from:
                pyo3::err::PyErr::new_type::h55eb3f4ff71d0b27 in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.7.rcgu.o)
            "_PyObject_GetAttr", referenced from:
                pyo3::types::any::PyAny::getattr::_$u7b$$u7b$closure$u7d$$u7d$::hd39fe11b70377580 in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.15.rcgu.o)
            "_PyGILState_Ensure", referenced from:
                pyo3::gil::prepare_freethreaded_python::_$u7b$$u7b$closure$u7d$$u7d$::finalize::hc67fcbad215bdfcb in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.10.rcgu.o)
                pyo3::gil::GILGuard::acquire::hc3ecf9e3d10079ec in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.10.rcgu.o)
            "_PyBytes_AsString", referenced from:
                pyo3::types::bytes::PyBytes::as_bytes::h83a377f562e02b60 in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.10.rcgu.o)
            "_Py_InitializeEx", referenced from:
                pyo3::gil::prepare_freethreaded_python::_$u7b$$u7b$closure$u7d$$u7d$::hf50c060ae8fa627a in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.10.rcgu.o)
            "_Py_Finalize", referenced from:
                pyo3::gil::prepare_freethreaded_python::_$u7b$$u7b$closure$u7d$$u7d$::finalize::hc67fcbad215bdfcb in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.10.rcgu.o)
            "_Py_IsInitialized", referenced from:
                pyo3::gil::prepare_freethreaded_python::_$u7b$$u7b$closure$u7d$$u7d$::hf50c060ae8fa627a in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.10.rcgu.o)
                pyo3::gil::prepare_freethreaded_python::_$u7b$$u7b$closure$u7d$$u7d$::finalize::hc67fcbad215bdfcb in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.10.rcgu.o)
            "_PyGILState_Release", referenced from:
                _$LT$pyo3..gil..GILGuard$u20$as$u20$core..ops..drop..Drop$GT$::drop::h4cad7b79d037c08b in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.10.rcgu.o)
            "_PyEval_SaveThread", referenced from:
                pyo3::gil::prepare_freethreaded_python::_$u7b$$u7b$closure$u7d$$u7d$::hf50c060ae8fa627a in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.10.rcgu.o)
            "_PyErr_Print", referenced from:
                pyo3::err::panic_after_error::h667c3f84912f4298 in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.7.rcgu.o)
            "_PyErr_GivenExceptionMatches", referenced from:
                pyo3::err::PyErr::is_instance::h9f9182db9db57990 in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.7.rcgu.o)
            "_PyList_Append", referenced from:
                pyo3::types::list::PyList::append::_$u7b$$u7b$closure$u7d$$u7d$::h04d9cd1923177461 in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.6.rcgu.o)
            "_PyExc_SystemError", referenced from:
                _$LT$pyo3..exceptions..PySystemError$u20$as$u20$pyo3..type_object..PyTypeInfo$GT$::type_object_raw::hf728ab47bcc04189 in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.0.rcgu.o)
            "_PyList_New", referenced from:
                pyo3::types::list::PyList::empty::ha21a4ae55595c619 in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.6.rcgu.o)
            "_PyExc_AttributeError", referenced from:
                _$LT$pyo3..exceptions..PyAttributeError$u20$as$u20$pyo3..type_object..PyTypeInfo$GT$::type_object_raw::hfd8de49ee1ff743c in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.0.rcgu.o)
            "__Py_Dealloc", referenced from:
                pyo3::ffi::object::Py_DECREF::h0420ac3aa4917e16 in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.4.rcgu.o)
            "_PyBytes_Size", referenced from:
                pyo3::types::bytes::PyBytes::as_bytes::h83a377f562e02b60 in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.10.rcgu.o)
            "_PyTuple_New", referenced from:
                pyo3::types::tuple::_$LT$impl$u20$pyo3..conversion..IntoPy$LT$pyo3..instance..Py$LT$pyo3..types..any..PyAny$GT$$GT$$u20$for$u20$$LP$A$C$$RP$$GT$::into_py::ha17853b045bd9d37 in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.4.rcgu.o)
                pyo3::types::tuple::_$LT$impl$u20$pyo3..conversion..IntoPy$LT$pyo3..instance..Py$LT$pyo3..types..any..PyAny$GT$$GT$$u20$for$u20$$LP$A$C$$RP$$GT$::into_py::hd2d23f9a99a76146 in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.12.rcgu.o)
            "_PyErr_NormalizeException", referenced from:
                pyo3::err::PyErr::normalized::h75e8127e42a18982 in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.7.rcgu.o)
            "_PyErr_PrintEx", referenced from:
                pyo3::err::PyErr::print::hb8f2f3ffe2d1fe30 in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.7.rcgu.o)
            "_PyErr_Restore", referenced from:
                pyo3::err::PyErr::restore::hb8e574d97e72ed24 in cryptography_rust.1jkgirnp9b8rrlvu.rcgu.o
                pyo3::err::PyErr::restore::hf5a4be04c1d3acb9 in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.7.rcgu.o)
            "_PyErr_Fetch", referenced from:
                pyo3::err::PyErr::fetch::h91002167fd75697f in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.7.rcgu.o)
            "_PyExc_TypeError", referenced from:
                _$LT$pyo3..exceptions..PyTypeError$u20$as$u20$pyo3..type_object..PyTypeInfo$GT$::type_object_raw::hec57ea5ed4e4d870 in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.0.rcgu.o)
            "_PyTuple_SetItem", referenced from:
                pyo3::types::tuple::_$LT$impl$u20$pyo3..conversion..IntoPy$LT$pyo3..instance..Py$LT$pyo3..types..any..PyAny$GT$$GT$$u20$for$u20$$LP$A$C$$RP$$GT$::into_py::ha17853b045bd9d37 in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.4.rcgu.o)
                pyo3::types::tuple::_$LT$impl$u20$pyo3..conversion..IntoPy$LT$pyo3..instance..Py$LT$pyo3..types..any..PyAny$GT$$GT$$u20$for$u20$$LP$A$C$$RP$$GT$::into_py::hd2d23f9a99a76146 in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.12.rcgu.o)
            "_PyEval_ThreadsInitialized", referenced from:
                pyo3::gil::prepare_freethreaded_python::_$u7b$$u7b$closure$u7d$$u7d$::hf50c060ae8fa627a in libpyo3-e06e690e5cecc294.rlib(pyo3-e06e690e5cecc294.pyo3.7kl564et-cgu.10.rcgu.o)
            "_PyModule_Create2", referenced from:
                pyo3::ffi::modsupport::PyModule_Create::hd8c2777f1ea7aa68 in cryptography_rust.36tytxljafqzsxvl.rcgu.o
          ld: symbol(s) not found for architecture x86_64
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
          

error: aborting due to previous error

error: could not compile `cryptography-rust`.

To learn more, run the command again with --verbose.
(cryptography) dpage@hal:~/git/cryptography/src/rust$ echo $CFLAGS
-mmacosx-version-min=10.12 -I/usr/local/openssl/include
(cryptography) dpage@hal:~/git/cryptography/src/rust$ echo $CXXFLAGS
-mmacosx-version-min=10.12
(cryptography) dpage@hal:~/git/cryptography/src/rust$ echo $LDFLAGS
-mmacosx-version-min=10.12 -L/usr/local/openssl/lib -L /Library/Frameworks/Python.framework -lPython
```
Any ideas? I'm completely unfamiliar with rust/cargo.

FYI, that was using the Python 3.9.1 distro from python.org (x86_64, not the Universal2 build), on an Intel mac running Big Sur.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -15,6 +15,7 @@
 ### Fixed
 - Set executable bit on shared library. [#110](https://github.com/PyO3/setuptools-rust/pull/110)
 - Don't require optional `wheel` dependency. [#111](https://github.com/PyO3/setuptools-rust/pull/111)
+- Set a more reasonable LC_ID_DYLIB entry on macOS. [#119](https://github.com/PyO3/setuptools-rust/pull/119)
 
 ## 0.11.6 (2020-12-13)
 

--- a/setuptools_rust/build.py
+++ b/setuptools_rust/build.py
@@ -180,6 +180,10 @@ class build_rust(RustCommand):
         if ext.native:
             rustflags += " -C target-cpu=native"
 
+        if not executable and sys.platform == "darwin":
+            ext_basename = os.path.basename(self.get_dylib_ext_path(ext, ext.name))
+            rustflags += f" -C link-args=-Wl,-install_name,@rpath/{ext_basename}"
+
         if rustflags:
             env["RUSTFLAGS"] = (env.get("RUSTFLAGS", "") + " " + rustflags).strip()
 
@@ -281,16 +285,7 @@ class build_rust(RustCommand):
 
                 ext.install_script(ext_path)
             else:
-                # Technically it's supposed to contain a
-                # `setuptools.Extension`, but in practice the only attribute it
-                # checks is `ext.py_limited_api`.
-                modpath = target_fname.split('.')[-1]
-                assert modpath not in build_ext.ext_map
-                build_ext.ext_map[modpath] = ext
-                try:
-                    ext_path = build_ext.get_ext_fullpath(target_fname)
-                finally:
-                    del build_ext.ext_map[modpath]
+                ext_path = self.get_dylib_ext_path(ext, target_fname)
 
             os.makedirs(os.path.dirname(ext_path), exist_ok=True)
             shutil.copyfile(dylib_path, ext_path)
@@ -307,7 +302,7 @@ class build_rust(RustCommand):
                     args.append(ext_path)
                     try:
                         output = subprocess.check_output(args, env=env)
-                    except subprocess.CalledProcessError as e:
+                    except subprocess.CalledProcessError:
                         pass
 
             # executables, win32(cygwin)-dll's, and shared libraries on
@@ -315,3 +310,16 @@ class build_rust(RustCommand):
             mode = os.stat(ext_path).st_mode
             mode |= (mode & 0o444) >> 2  # copy R bits to X
             os.chmod(ext_path, mode)
+
+    def get_dylib_ext_path(self, ext, target_fname):
+        build_ext = self.get_finalized_command("build_ext")
+        # Technically it's supposed to contain a
+        # `setuptools.Extension`, but in practice the only attribute it
+        # checks is `ext.py_limited_api`.
+        modpath = target_fname.split('.')[-1]
+        assert modpath not in build_ext.ext_map
+        build_ext.ext_map[modpath] = ext
+        try:
+            return build_ext.get_ext_fullpath(target_fname)
+        finally:
+            del build_ext.ext_map[modpath]

